pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotClusterConfigs.java

```diff
@@ -0,0 +1,123 @@
+import java.util.List;
+import java.util.Map;
+import javax.inject.Inject;
+import javax.ws.rs.DELETE;
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import org.apache.helix.HelixAdmin;
+import org.apache.helix.model.HelixConfigScope;
+import org.apache.helix.model.builder.HelixConfigScopeBuilder;
+import org.apache.pinot.controller.helix.core.PinotHelixResourceManager;
+import org.apache.pinot.spi.utils.JsonUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+@Api(tags = Constants.CLUSTER_TAG)
+@Path("/")
+public class PinotClusterConfigs {
+  private static final Logger LOGGER = LoggerFactory.getLogger(PinotClusterConfigs.class);
+
+  @Inject
+  PinotHelixResourceManager pinotHelixResourceManager;
+
+  @GET
+  @Path("/cluster/configs")
+  @Produces(MediaType.APPLICATION_JSON)
+  @ApiOperation(value = "List cluster configurations", notes = "List cluster level configurations")
+  @ApiResponses(value = {@ApiResponse(code = 200, message = "Success"), @ApiResponse(code = 500, message = "Internal server error")})
+  public String listClusterConfigs() {
+    HelixAdmin helixAdmin = pinotHelixResourceManager.getHelixAdmin();
+    HelixConfigScope configScope = new HelixConfigScopeBuilder(HelixConfigScope.ConfigScopeProperty.CLUSTER)
+        .forCluster(pinotHelixResourceManager.getHelixClusterName()).build();
+    List<String> configKeys = helixAdmin.getConfigKeys(configScope);
+    ObjectNode ret = JsonUtils.newObjectNode();
+    Map<String, String> configs = helixAdmin.getConfig(configScope, configKeys);
+    for (String key : configs.keySet()) {
+      ret.put(key, configs.get(key));
+    }
+    return ret.toString();
+  }
+
+  @POST
+  @Path("/cluster/configs")
+  @ApiOperation(value = "Update cluster configuration")
+  @Produces(MediaType.APPLICATION_JSON)
+  @ApiResponses(value = {@ApiResponse(code = 200, message = "Success"), @ApiResponse(code = 500, message = "Server error updating configuration")})
+  public SuccessResponse updateClusterConfig(String body) {
+    try {
+      JsonNode jsonNode = JsonUtils.stringToJsonNode(body);
+      HelixAdmin admin = pinotHelixResourceManager.getHelixAdmin();
+      HelixConfigScope configScope = new HelixConfigScopeBuilder(HelixConfigScope.ConfigScopeProperty.CLUSTER)
+          .forCluster(pinotHelixResourceManager.getHelixClusterName()).build();
+      Iterator<String> fieldNamesIterator = jsonNode.fieldNames();
+      while (fieldNamesIterator.hasNext()) {
+        String key = fieldNamesIterator.next();
+        String value = jsonNode.get(key).textValue();
+        admin.setConfig(configScope, Collections.singletonMap(key, value));
+      }
+      return new SuccessResponse("Updated cluster config.");
+    } catch (IOException e) {
+      String errStr = "Error converting request to cluster config.";
```

Comment:
Any reason for using the `errStr` variable, as opposed to in-lining the message?

Reply:
Copy pasted template from other file, updated it.